_posts/en/newsletters/2020-09-02-newsletter.md

```diff
@@ -0,0 +1,252 @@
+  [_Default signet discussion_ news item](#default-signet-discussion).
+  ([transcript][signet transcript], [video][signet video])
+
+- **Bitcoin Core GUI meeting**: Anonymized participants, including designers and
+  developers, met to discuss the Bitcoin Core graphical user interface, its
+  current state, how it could be improved, and the constraints that
+  any possible revamp would be subject to. For example, one change that has been
+  discussed in the past is moving from Qt Widgets to the Qt QML framework which
+  is more flexible and easily customized. ([transcript][bitcoin core gui
+  transcript]).
+
+- **Sydney meetup discussion** Ruben Somsen and Rusty Russell participated in a
+  discussion on statechains, upgrading LN channels, and [lnprototest][]. Somsen
+  outlined how statechains (see [Newsletter #100][news100 statechains]) could
+  potentially be used to swap out your Lightning channel counterparty or, in the
+  case of DLCs (see [Newsletter #81][news81 dlc]), swap out a portion of your
+  position without closing the channel. With various possible Lightning channel
+  upgrades on the horizon, participants speculated about which upgrades could use dynamic commitments
+  (see [Newsletter #108][news108 dynamic commitments]) and which would require
+  [splicing][topic splicing]. Finally, Russell explained how the Lightning
+  protocol test suite lnprototest could be used to find bugs in existing
+  implementations and assist with ensuring interoperability across
+  implementations when building out new features.  ([transcript][sydney
+  transcript]).
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1 repo],
+[Hardware Wallet Interface (HWI)][hwi], [Bitcoin Improvement Proposals
+(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #19797][] strays from Satoshi's original vision™ by
+  removing an obsolete validity check for IPv6 addresses constructed by
+  Bitcoin (Core) 0.2.8 and earlier.  Those early nodes had a bug that
+  would construct malformed addresses.
+  This change should have no impact on the current P2P network, as
+  affected versions are unable to communicate with current software due
+  to the introduction of checksums for P2P messages in Bitcoin
+  (Core) 0.2.9, which were later made mandatory.
+
+- [Bitcoin Core #19731][] extends the `getpeerinfo` RPC output with two new
+  fields: `last_block` and `last_transaction`. `last_block` is the time that
+  the node most recently received a block from the peer which it had not seen
+  before and which passed preliminary validity checks. `last_transaction` is the
+  last time that the node received a transaction from the peer that it
+  accepted to its mempool. Both of these metrics are used when choosing peers
```

Comment:
These sentences each have an ambiguous it ("it had not" and "that it accepted"). Suggestion:

`last_block` is the time that the node most recently received a block it had not seen before from the peer and which passed preliminary validity checks. `last_transaction` is the time that the node most recently received a transaction that it accepted into its mempool from the peer.

Reply:
Making a different change to hopefully make it clear that it has to be a novel block or transaction, rather than a block or tx the node already received from another peer.

Reply:
s/`last_transaction` is the last time that the peer sent the local node a transaction that the node accepted to its mempool./`last_transaction` is the time that the peer sent the local node the most recent valid transaction./

is valid = accepted to mempool?

Reply:
Only valid transactions are accepted into the mempool, but a transaction can be valid but not accepted into the mempool (e.g., it pays too low of a feerate or there's a copy of the transaction already in the mempool).

Reply:
It's the time of the most recent transaction that we received and accepted into our mempool from that peer. See `src/net_processing.cpp::L3022` in master, where `ProcessMessage()` handles TX net messages and this value is set with `pfrom.nLastTXTime = GetTime()`.

Reply:
The conditional immediately preceding it checks that we have not already seen it and that it is accepted into our mempool:

```cpp
If (!AlreadyHave(CInv(MSG_WTX, wtxid), m_mempool) &&
    AcceptToMemoryPool(m_mempool, state, ptx, &lRemovedTxn, false /* bypass_limits */, 0 /* nAbsurdFee */)) {
```